### DISCUSSION TITLE
Supabase bucket - new row violates row-level security policy for table "objects"

### DISCUSSION THREAD

**Discussion by Kroplewski-M** [supabase/supabase#10059] [Questions] [answered]
Hi, I am having an issue with uploading avatars to my supabase bucket called avatars as its giving me "new row violates row-level security policy for table "objects"". I tried StackOverflow solutions and nothing. Before trying to upload I log in using supabse so my user is authenticated yet its still not letting me upload. I added this policy in storage.objects:
``` (role() = 'authenticated'::text)```

and clicked the insert button. Does anyone know what I am doing wrong? I assume its something to do with the policies. Thanks
this is how I'm trying to upload my avatar:

``` 
try{
                const { data, error } = await supabase
                    .storage
                    .from('/public/avatars')
                    .upload(`${values.email}.png`, values.avatar, {
                        cacheControl: '3600',
                        upsert: true
                });
                if(error) throw error;
            }catch(error){
                console.log(error);
            }
```

**Comment by GaryAustin1** [comment, marked as answer]
The bucket name probably should not include `/public/` and just the bucket name `avatars`.

If that is not it, then you could set RLS to `true` to rule out having a signed in user as the problem.

**Reply by Kroplewski-M** [reply]
I had the name as only ```avatars``` previously and it didn't work. When you say set the RLS, where do you mean to set it? I set the bucket to both public and private and both didn't work. 
I get a link with the error and when I open it I get this:
``` {"statusCode":"400","error":"Bad Request","message":"headers must have required property 'authorization'"} ```
Even though I'm logged in, do you know the issue? thanks

**Reply by GaryAustin1** [reply]
RLS is referring to your storage.objects policy.  A public bucket works the same for all operations except has a public URL available as a private bucket, so still needs to meet storage RLS policies.  By temporarily setting it to `true` instead of `role()=....` then any user could upload.  If that works then you don't really have a signed in user when you make the storage call.

I have no idea what you are referring to by get "link with the error", so not sure what that is doing.

**Reply by Kroplewski-M** [reply]
i just set it to true like you said and it worked ill just leave it as that.  Thanks
